src/alfasim_sdk/alfacase/generate_schema.py

```diff
@@ -0,0 +1,388 @@
+
+    Regex from: https://github.com/jpvanhal/inflection/blob/0.5.1/inflection/__init__.py#L397
+    """
+    import re
+
+    value = re.sub(r"([A-Z]+)([A-Z][a-z])", r"\1_\2", value)
+    value = re.sub(r"([a-z\d])([A-Z])", r"\1_\2", value)
+    value = value.replace("-", "_")
+    return value.lower()
+
+
+def obtain_schema_name(class_: type) -> str:
+    """Obtains a schema name from TestCase class"""
+    return convert_to_snake_case(class_.__name__) + "_schema"
+
+
+def _get_attr_name(key: str, value: attr.ib) -> str:
+    """
+    Note: for stricyyaml schema, an Optional type means that the user don't need
+    to inform a value. While for type hint, an Optional type means that the attribute accepts None.
+
+    Therefore, if the given value has a default, it means a "Optional" type for strictyaml
+    """
+    return f'"{key}"' if value.default is attr.NOTHING else f'Optional("{key}")'
+
+
+# plugins:
+#   For now, Plugins inputs (GUI) will not be able to be customizable by YAML
+
+# initial_condition_arrays:
+#   Only used for tests, doesnt make sense to allows YAML to configure this (for now).
+
+IGNORED_PROPERTIES = (
+    "plugins",
+    "initial_condition_arrays",
+    "nonsplit_equipment",
+    "table_parameters",
+)
+
+
+def _get_attribute_schema(
+    key: str, value: attr.ib, indentation=INDENTANTION * 2
+) -> str:
+    """
+    Helper method that return the equivalent schema for the given key and value.
+
+    Keep in mind that for stricyyaml schema, an Optional type means that the user don't need
+    to inform a value. While for type hint, an Optional type means that the attribute accepts None.
+
+
+    """
+    if value.default is attr.NOTHING and is_optional_type(value.type):
+        msg = (
+            "StrictYAML doesn't support None value (only missing keys denote a value), "
+            "therefore Optional type are only allowed when the case has a default value."
+        )
+        raise TypeError(msg)
+    attribute_name = _get_attr_name(key, value)
+    attribute_value = _get_attr_value(
+        value.type, block_indentation=indentation, key=key
+    )
+
+    return indentation + f"{attribute_name}: {attribute_value}" + ","
+
+
+def generate_alfacase_schema(class_: type) -> str:
+    """
+    Return a schema for the given `class_` that is decorated with @attr and has type hints
+    """
+    lines = [f"{obtain_schema_name(class_)} = Map(", f"{INDENTANTION}{{"]
+    lines.extend(
+        _get_attribute_schema(key, value)
+        for key, value in attr.fields_dict(class_).items()
+        if key not in IGNORED_PROPERTIES
+    )
+    lines.extend([f"{INDENTANTION}}}", ")", ""])
+
+    return "\n".join(lines)
+
+
+def _is_from_typing_module(type_: type) -> bool:
+    """Helper method to verify if the given `type_` is instance of List/Dict/Union"""
+    return is_list(type_) or is_dict(type_) or typing_inspect.is_union_type(type_)
+
+
+def _obtain_referred_type(type_) -> List[type]:
```

Comment:
Is it just for me or the code below this line isn't syntax-higligthed? weeeird
(I'm not talking about the comments, but the code itself)

Reply:
Here too

Reply:
Yep, me too everything bellow `_is_from_typing_module` doesn't has highlighted on Github 👀